mlrun/feature_store/feature_set.py

```diff
@@ -391,6 +403,52 @@ def purge_targets(self, target_names: List[str] = None, silent: bool = False):
             self, mlrun.api.schemas.AuthorizationAction.delete
         )
 
+        purge_targets = self._reload_and_get_status_targets(
+            target_names=target_names, silent=silent
+        )
+
+        if purge_targets:
+            # TODO - benb - run_uuid shouldn't be a part of purge - remove logic after toggle implementation
+            from mlrun.datastore.targets import generate_target_run_uuid
+
+            run_uuid = generate_target_run_uuid()
+            purge_target_names = list(purge_targets.keys())
+            for target_name in purge_target_names:
+                target = purge_targets[target_name]
+                driver = get_target_driver(target_spec=target, resource=self)
+                try:
+                    driver.purge()
+                    driver.run_uuid = run_uuid
+                except FileNotFoundError:
+                    pass
+                del self.status.targets[target_name]
+
+            self.save()
+
+    def update_targets_for_ingest(
+        self,
+        targets: List[DataTargetBase],
+        overwrite: bool = None,
+    ):
+        ingestion_target_names = [t if isinstance(t, str) else t.name for t in targets]
```

Comment:
But `targets` is of type `List[DataTargetBase]`. Why are you checking it `t` is a `str`? Is this a leftover from a time when strings were accepted here?